MBBSEmu/HostProcess/MbbsHost.cs

```diff
@@ -1297,22 +1331,54 @@ private void PatchRelocation(MbbsModule module)
             }
         }
 
+        /// <summary>
+        ///     Marks the specified module as "Enabled" and executes the _INIT_ routine
+        /// </summary>
+        /// <param name="moduleId"></param>
         private void EnableModule(string moduleId)
         {
             _modules[moduleId].ModuleConfig.ModuleEnabled = true;
         }
 
-        private void DisableModule(string moduleId)
+        /// <summary>
+        ///     Disables a Module while the service host is running
+        /// </summary>
+        /// <param name="moduleId"></param>
+        /// <param name="isCrashed"></param>
+        private void DisableModule(string moduleId, bool isCrashed = false)
         {
-            if (_channelDictionary.Values.Where(session => session.SessionState == EnumSessionState.InModule).Any(channel => channel.CurrentModule.ModuleIdentifier == moduleId))
+            //Ensure Module is marked Disabled
+            _modules[moduleId].ModuleConfig.ModuleEnabled = false;
+
+            //Notify Users and Exit Modules
+            foreach (var c in _channelDictionary.Values.Where(x => x.CurrentModule.ModuleIdentifier == moduleId))
             {
-                Logger.Warn($"(Sysop Command) Tried to disable {moduleId} while in use -- Wait until all users have exited module");
-                return;
+                if (isCrashed)
+                {
+                    c.SendToClient($"|RESET|\r\n|B||RED|The module you were in ({moduleId}) has crashed. Please contact the Sysop and try again later.|RESET|\r\n".EncodeToANSIArray());
+                }
+                else
+                {
+                    c.SendToClient($"|RESET|\r\n|B||RED|The module you were in ({moduleId}) has been disabled by the Sysop. Please try again later.|RESET|\r\n".EncodeToANSIArray());
+                }
+
+                ExitModule(c);
             }
 
-            _modules[moduleId].ModuleConfig.ModuleEnabled = false;
+            if (isCrashed)
+            {
+                Logger.Error($"Module {moduleId} has crashed. Disabling.");
+            }
+            else
+            {
+                Logger.Info($"Module {moduleId} has been disabled by /SYSOP command.");
+            }
```

Comment:
probably want to move this before the call to `ExitModule(c)` in crash `ExitModule` bombs on its own?